cannot create cluster on existing VPC with config file

**What happened?**
when i created cluster on exisiting VPC with config files, it failed.

**What you expected to happen?**

**How to reproduce it?**
```
$ eksctl create cluster -f cluster.yaml
[ℹ]  using region ap-northeast-1
[ℹ]  setting availability zones to [ap-northeast-1a ap-northeast-1c ap-northeast-1d]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1d41fb8]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/vpc.SetSubnets(0xc00028c000, 0xc00028c000, 0x0)
        /go/src/github.com/weaveworks/eksctl/pkg/vpc/vpc.go:24 +0x148
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster.func3(0x3042e60, 0xc000a21500)
        /go/src/github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:301 +0x8d7
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster(0xc000674080, 0xc00028c000, 0x0, 0x0, 0xc0006fc780, 0x4, 0x329d123)
        /go/src/github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:361 +0x5dc
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmd.func1(0xc0006fc780, 0xc000678ce0, 0x0, 0x2)
        /go/src/github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:53 +0x94
github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra.(*Command).execute(0xc0006fc780, 0xc000678ca0, 0x2, 0x2, 0xc0006fc780, 0xc000678ca0)
        /go/src/github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x5154c80, 0x1, 0x5b, 0x2b)
        /go/src/github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra.(*Command).Execute(0x5154c80, 0x0, 0xc0000ac2f0)
        /go/src/github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /go/src/github.com/weaveworks/eksctl/cmd/eksctl/main.go:65 +0x2d
```

- config.yaml
``` 
apiVersion: eksctl.io/v1alpha3
kind: ClusterConfig

metadata:
  name: cluster-in-existing-vpc
  region: ap-northeast-1

vpc:
  subnets:
    private:
      ap-northeast-1a: {id: subnet-099ab88b723e0xxxx}
      ap-northeast-1c: {id: subnet-0b28a3843856dxxxx}
      ap-northeast-1d: {id: subnet-0c2439e34d6a3xxxx}

nodeGroups:
  - name: ng-1-workers
    instanceType: m5.xlarge
    desiredCapacity: 10
    privateNetworking: true
```

**Anything else we need to know?**

**Versions**
```
$ eksctl version 
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.1.18"}
$ uname -a 
Darwin sasaki-alpaca.local 18.2.0 Darwin Kernel Version 18.2.0: Mon Nov 12 20:24:46 PST 2018; root:xnu-4903.231.4~2/RELEASE_X86_64 x86_64
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
```


**Logs**

There were 2 results.

(1) `private` -> `Private`
```
vpc:
  subnets:
    Private:
      ap-northeast-1a: {Id: subnet-099ab88b723e0xxxx}
      ap-northeast-1c: {Id: subnet-0b28a3843856dxxxx}
      ap-northeast-1d: {Id: subnet-0c2439e34d6a3xxxx}
```

result
```
$ eksctl create cluster -f cluster.yaml
[ℹ]  using region ap-northeast-1
[✔]  using existing VPC () and subnets (private:[  ] public:[])
[!]  custom VPC/subnets will be used; if resulting cluster doesn't function as expected, make sure to review the configuration of VPC/subnets
[ℹ]  creating EKS cluster "cluster-in-existing-vpc" in "ap-northeast-1" region
[ℹ]  will create a CloudFormation stack for cluster itself and 0 nodegroup stack(s)
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=ap-northeast-1 --name=cluster-in-existing-vpc'
[ℹ]  creating cluster stack "eksctl-cluster-in-existing-vpc-cluster"
panic: runtime error: index out of range

goroutine 62 [running]:
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).addResourcesForVPC(0xc0000b44b0)
        /go/src/github.com/weaveworks/eksctl/pkg/cfn/builder/vpc.go:70 +0xa11
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).AddAllResources(0xc0000b44b0, 0xc0000b44b0, 0xc000b0bee0)
        /go/src/github.com/weaveworks/eksctl/pkg/cfn/builder/cluster.go:52 +0x166
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).CreateCluster(0xc0005e7620, 0xc00068fda0, 0x0, 0x0, 0x2dde380, 0xc000057748)
        /go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/cluster.go:17 +0x30c
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).CreateCluster-fm(0xc00068fda0, 0x0, 0x0, 0x1, 0x1)
        /go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:64 +0x48
github.com/weaveworks/eksctl/pkg/cfn/manager.Run.func1(0xc0009de330, 0xc0009b9040, 0x1, 0x1, 0xc000c37310, 0x0)
        /go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:32 +0x11f
created by github.com/weaveworks/eksctl/pkg/cfn/manager.Run
        /go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:28 +0xa7
```

private is empty?
```
using existing VPC () and subnets (private:[  ] public:[])
```

(2) `public` -> `Public`
```
vpc:
  subnets:
    Public:
      ap-northeast-1a: {id: subnet-06e6ace6301e9xxxx}
      ap-northeast-1c: {id: subnet-092d52f6f5805xxxx}
      ap-northeast-1d: {id: subnet-07ecc8c96d5f5xxxx}
```

result
```
$ eksctl create cluster -f cluster.yaml
[ℹ]  using region ap-northeast-1
[✔]  using existing VPC () and subnets (private:[] public:[subnet-06e6ace6301e9xxxx subnet-092d52f6f5805xxxx subnet-07ecc8c96d5f5xxxx])
[!]  custom VPC/subnets will be used; if resulting cluster doesn't function as expected, make sure to review the configuration of VPC/subnets
[ℹ]  creating EKS cluster "cluster-in-existing-vpc" in "ap-northeast-1" region
[ℹ]  will create a CloudFormation stack for cluster itself and 0 nodegroup stack(s)
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=ap-northeast-1 --name=cluster-in-existing-vpc'
[ℹ]  creating cluster stack "eksctl-cluster-in-existing-vpc-cluster"
[✖]  unexpected status "ROLLBACK_IN_PROGRESS" while waiting for CloudFormation stack "eksctl-cluster-in-existing-vpc-cluster" to reach "CREATE_COMPLETE" status
[ℹ]  fetching stack events in attempt to troubleshoot the root cause of the failure
[ℹ]  AWS::CloudFormation::Stack/eksctl-cluster-in-existing-vpc-cluster: ROLLBACK_IN_PROGRESS – "The following resource(s) failed to create: [InternetGateway, ServiceRole, NATIP, VPC]. . Rollback requested by user."
[✖]  AWS::EC2::EIP/NATIP: CREATE_FAILED – "Resource creation cancelled"
[✖]  AWS::IAM::Role/ServiceRole: CREATE_FAILED – "Resource creation cancelled"
[✖]  AWS::EC2::InternetGateway/InternetGateway: CREATE_FAILED – "Resource creation cancelled"
[ℹ]  AWS::EC2::EIP/NATIP: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::IAM::Role/ServiceRole: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::EC2::InternetGateway/InternetGateway: CREATE_IN_PROGRESS – "Resource creation Initiated"
[✖]  AWS::EC2::VPC/VPC: CREATE_FAILED – "Value () for parameter cidrBlock is invalid. This is not a valid CIDR block. (Service: AmazonEC2; Status Code: 400; Error Code: InvalidParameterValue; Request ID: 3592b865-7524-49d3-8626-acdcae63b611)"
[ℹ]  AWS::EC2::EIP/NATIP: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::InternetGateway/InternetGateway: CREATE_IN_PROGRESS
[ℹ]  AWS::IAM::Role/ServiceRole: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::VPC/VPC: CREATE_IN_PROGRESS
[ℹ]  AWS::CloudFormation::Stack/eksctl-cluster-in-existing-vpc-cluster: CREATE_IN_PROGRESS – "User Initiated"
[ℹ]  1 error(s) occurred and cluster hasn't been created properly, you may wish to check CloudFormation console
[ℹ]  to cleanup resources, run 'eksctl delete cluster --region=ap-northeast-1 --name=cluster-in-existing-vpc'
[✖]  waiting for CloudFormation stack "eksctl-cluster-in-existing-vpc-cluster" to reach "CREATE_COMPLETE" status: ResourceNotReady: failed waiting for successful resource state
[✖]  failed to create cluster "cluster-in-existing-vpc"
```

```
AWS::EC2::VPC/VPC: CREATE_FAILED – "Value () for parameter cidrBlock is invalid. This is not a valid CIDR block.
```

when i checked CFn Template on Management Console, VPC CiderBlock was `""`.
```
		"VPC": {
			"Type": "AWS::EC2::VPC",
			"Properties": {
				"CidrBlock": "",
				"EnableDnsHostnames": true,
				"EnableDnsSupport": true,
				"Tags": [
					{
						"Key": "Name",
						"Value": {
							"Fn::Sub": "${AWS::StackName}/VPC"
						}
					}
				]
			}
		},
```

--- a/examples/04-existing-vpc.yaml
+++ b/examples/04-existing-vpc.yaml
@@ -8,19 +8,25 @@ metadata:
   region: eu-north-1
 
 vpc:
+  id: "vpc-0dd338ecf29863c55"  # (optional, must match VPC ID used for each subnet below)
+  cidr: "192.168.0.0/16"       # (optional, must match CIDR used by the given VPC)
   subnets:
-    private:
-      # NOTE: you cannot run this example directly,
-      # you will need to set actual subenet IDs here
-      eu-north-1a:
-        id: subnet-a123456
-      eu-north-1b:
-        id: subnet-b123456
-      eu-north-1c:
-        id: subnet-c123456
+    # must provide 'Private' and/or 'Public' subnets by availibility zone as shown
+    Private:
+      us-west-2a:
+        id: "subnet-0b2512f8c6ae9bf30"
+        cidr: "192.168.128.0/19" # (optional, must match CIDR used by the given subnet)
+
+      us-west-2b:
+        id: "subnet-08cb9a2ed60394ce3"
+        cidr: "192.168.64.0/19"  # (optional, must match CIDR used by the given subnet)
+
+      us-west-2c:
+        id: "subnet-00f71956cdec8f1dc"
+        cidr: "192.168.0.0/19"   # (optional, must match CIDR used by the given subnet)
 
 nodeGroups:
   - name: ng-1
     instanceType: m5.xlarge
     desiredCapacity: 2
-    privateNetworking: true
+    privateNetworking: true # if only 'Private' subnets are given, this must be enabled

--- a/pkg/apis/eksctl.io/v1alpha3/vpc.go
+++ b/pkg/apis/eksctl.io/v1alpha3/vpc.go
@@ -77,7 +77,7 @@ func (c *ClusterConfig) SubnetIDs(topology SubnetTopology) []string {
 }
 
 // ImportSubnet loads a given subnet into cluster config
-func (c *ClusterConfig) ImportSubnet(topology SubnetTopology, az, subnetID, cidr string) {
+func (c *ClusterConfig) ImportSubnet(topology SubnetTopology, az, subnetID, cidr string) error {
 	subnetCIDR, _ := ipnet.ParseCIDR(cidr)
 
 	if c.VPC.Subnets == nil {
@@ -89,10 +89,19 @@ func (c *ClusterConfig) ImportSubnet(topology SubnetTopology, az, subnetID, cidr
 	if network, ok := c.VPC.Subnets[topology][az]; !ok {
 		c.VPC.Subnets[topology][az] = Network{ID: subnetID, CIDR: subnetCIDR}
 	} else {
-		network.ID = subnetID
-		network.CIDR = subnetCIDR
+		if network.ID == "" {
+			network.ID = subnetID
+		} else if network.ID != subnetID {
+			return fmt.Errorf("subnet ID %q is not the same as %q", network.ID, subnetID)
+		}
+		if network.CIDR == nil {
+			network.CIDR = subnetCIDR
+		} else if network.CIDR.String() != subnetCIDR.String() {
+			return fmt.Errorf("subnet CIDR %q is not the same as %q", network.CIDR.String(), subnetCIDR.String())
+		}
 		c.VPC.Subnets[topology][az] = network
 	}
+	return nil
 }
 
 // HasSufficientPublicSubnets validates if there is a sufficient

--- a/pkg/cfn/builder/api_test.go
+++ b/pkg/cfn/builder/api_test.go
@@ -8,6 +8,7 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/aws/aws-sdk-go/aws"
 	cfn "github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/stretchr/testify/mock"
@@ -241,6 +242,16 @@ var _ = Describe("CloudFormation template builder API", func() {
 			return strings.Join(ids, ",") == compare
 		}
 		testVPC := testVPC()
+
+		p.MockEC2().On("DescribeVpcs", mock.MatchedBy(func(input *ec2.DescribeVpcsInput) bool {
+			return *input.VpcIds[0] == vpcID
+		})).Return(&ec2.DescribeVpcsOutput{
+			Vpcs: []*ec2.Vpc{{
+				VpcId:     aws.String(vpcID),
+				CidrBlock: aws.String("192.168.0.0/16"),
+			}},
+		}, nil)
+
 		for t := range subnetLists {
 			func(list string, subnetsByAz map[string]api.Network) {
 				subnets := strings.Split(list, ",")

--- a/pkg/cfn/builder/cluster.go
+++ b/pkg/cfn/builder/cluster.go
@@ -108,11 +108,11 @@ func (c *ClusterResourceSet) GetAllOutputs(stack cfn.Stack) error {
 	c.spec.VPC.SecurityGroup = c.outputs[CfnOutputClusterSecurityGroup]
 	c.spec.VPC.SharedNodeSecurityGroup = c.outputs[CfnOutputClusterSharedNodeSecurityGroup]
 
-	if err := vpc.UseSubnets(c.provider, c.spec, api.SubnetTopologyPrivate, strings.Split(c.outputs[CfnOutputClusterSubnetsPrivate], ",")); err != nil {
+	if err := vpc.UseSubnetsFromList(c.provider, c.spec, api.SubnetTopologyPrivate, strings.Split(c.outputs[CfnOutputClusterSubnetsPrivate], ",")); err != nil {
 		return err
 	}
 
-	if err := vpc.UseSubnets(c.provider, c.spec, api.SubnetTopologyPublic, strings.Split(c.outputs[CfnOutputClusterSubnetsPublic], ",")); err != nil {
+	if err := vpc.UseSubnetsFromList(c.provider, c.spec, api.SubnetTopologyPublic, strings.Split(c.outputs[CfnOutputClusterSubnetsPublic], ",")); err != nil {
 		return err
 	}
 

--- a/pkg/ctl/create/cluster.go
+++ b/pkg/ctl/create/cluster.go
@@ -36,7 +36,6 @@ var (
 
 	kopsClusterNameForVPC string
 	subnets               map[api.SubnetTopology]*[]string
-	subnetsGiven          bool
 	addonsStorageClass    bool
 	withoutNodeGroup      bool
 )
@@ -114,11 +113,11 @@ func checkEachNodeGroup(cfg *api.ClusterConfig, check func(i int, ng *api.NodeGr
 	return nil
 }
 
-func doCheckSubnetsGivenAsConfigFile(cfg *api.ClusterConfig) bool {
+func checkSubnetsGiven(cfg *api.ClusterConfig) bool {
 	return len(cfg.VPC.Subnets[api.SubnetTopologyPrivate])+len(cfg.VPC.Subnets[api.SubnetTopologyPublic]) != 0
 }
 
-func doCheckSubnetsGivenAsFlags() bool {
+func checkSubnetsGivenAsFlags() bool {
 	return len(*subnets[api.SubnetTopologyPrivate])+len(*subnets[api.SubnetTopologyPublic]) != 0
 }
 
@@ -167,6 +166,7 @@ func doCreateCluster(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg stri
 			"vpc-private-subnets",
 			"vpc-public-subnets",
 			"vpc-cidr",
+			"vpc-from-kops-cluster",
 		}
 
 		for _, f := range incompatibleFlags {
@@ -189,9 +189,7 @@ func doCreateCluster(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg stri
 			cfg.VPC = api.NewClusterVPC()
 		}
 
-		subnetsGiven = doCheckSubnetsGivenAsConfigFile(cfg)
-
-		if subnetsGiven && len(cfg.AvailabilityZones) != 0 {
+		if checkSubnetsGiven(cfg) && len(cfg.AvailabilityZones) != 0 {
 			return fmt.Errorf("vpc.subnets and availabilityZones cannot be set at the same time")
 		}
 
@@ -237,8 +235,6 @@ func doCreateCluster(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg stri
 		if cfg.Status != nil {
 			return fmt.Errorf("status fields are read-only")
 		}
-
-		subnetsGiven = doCheckSubnetsGivenAsFlags()
 
 		if withoutNodeGroup {
 			cfg.NodeGroups = nil
@@ -292,6 +288,16 @@ func doCreateCluster(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg stri
 		}
 		kubeconfigPath = kubeconfig.AutoPath(meta.Name)
 	}
+
+	if checkSubnetsGivenAsFlags() {
+		// load subnets from local map created from flags, into the config
+		for topology := range subnets {
+			if err := vpc.UseSubnetsFromList(ctl.Provider, cfg, topology, *subnets[topology]); err != nil {
+				return err
+			}
+		}
+	}
+	subnetsGiven := checkSubnetsGiven(cfg) // this will be false when neither flags nor config has any subnets
 
 	createOrImportVPC := func() error {
 
@@ -354,10 +360,8 @@ func doCreateCluster(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg stri
 			return fmt.Errorf("--vpc-private-subnets/--vpc-public-subnets and --zones %s", cmdutils.IncompatibleFlags)
 		}
 
-		for topology := range subnets {
-			if err := vpc.UseSubnets(ctl.Provider, cfg, topology, *subnets[topology]); err != nil {
-				return err
-			}
+		if err := vpc.UseSubnets(ctl.Provider, cfg); err != nil {
+			return err
 		}
 
 		if err := cfg.HasSufficientSubnets(); err != nil {

--- a/pkg/eks/eks.go
+++ b/pkg/eks/eks.go
@@ -133,7 +133,7 @@ func (c *ClusterProvider) GetClusterVPC(spec *api.ClusterConfig, ignoreMissingKe
 		// either of subnet topologies are optional
 		if subnets, ok := outputs[builder.CfnOutputClusterSubnets+string(topology)]; ok {
 			subnets := strings.Split(subnets, ",")
-			if err := vpc.UseSubnets(c.Provider, spec, topology, subnets); err != nil {
+			if err := vpc.UseSubnetsFromList(c.Provider, spec, topology, subnets); err != nil {
 				return err
 			}
 		}

--- a/pkg/vpc/vpc.go
+++ b/pkg/vpc/vpc.go
@@ -84,11 +84,23 @@ func ImportVPC(provider api.ClusterProvider, spec *api.ClusterConfig, id string)
 	if err != nil {
 		return err
 	}
-	spec.VPC.ID = *vpc.VpcId
-	spec.VPC.CIDR, err = ipnet.ParseCIDR(*vpc.CidrBlock)
-	if err != nil {
-		return err
+	if spec.VPC.ID == "" {
+		spec.VPC.ID = *vpc.VpcId
+	} else if spec.VPC.ID != *vpc.VpcId {
+		return fmt.Errorf("VPC ID %q is the same as not %q", spec.VPC.ID, *vpc.VpcId)
 	}
+	if spec.VPC.CIDR == nil {
+		spec.VPC.CIDR, err = ipnet.ParseCIDR(*vpc.CidrBlock)
+		if err != nil {
+			return err
+		}
+	} else if cidr := spec.VPC.CIDR.String(); cidr != *vpc.CidrBlock {
+		return fmt.Errorf("VPC CIDR block %q is not the same as %q",
+			cidr,
+			*vpc.CidrBlock,
+		)
+	}
+
 	return nil
 }
 
@@ -96,32 +108,61 @@ func ImportVPC(provider api.ClusterProvider, spec *api.ClusterConfig, id string)
 // it will use provider to call describeVPC based on the VPC ID of the
 // first subnet; all subnets must be in the same VPC
 func ImportSubnets(provider api.ClusterProvider, spec *api.ClusterConfig, topology api.SubnetTopology, subnets []*ec2.Subnet) error {
+	if spec.VPC.ID != "" {
+		// ensure VPC gets imported and validated firt, if it's already set
+		if err := ImportVPC(provider, spec, spec.VPC.ID); err != nil {
+			return err
+		}
+	}
 	for _, subnet := range subnets {
 		if spec.VPC.ID == "" {
+			// if VPC wasn't defined, import it based on VPC of the first
+			// subnet that we have
 			if err := ImportVPC(provider, spec, *subnet.VpcId); err != nil {
 				return err
 			}
-		} else if spec.VPC.ID != *subnet.VpcId {
+		} else if spec.VPC.ID != *subnet.VpcId { // be sure to use the same VPC
 			return fmt.Errorf("given %s is in %s, not in %s", *subnet.SubnetId, *subnet.VpcId, spec.VPC.ID)
 		}
 
-		spec.ImportSubnet(topology, *subnet.AvailabilityZone, *subnet.SubnetId, *subnet.CidrBlock)
+		if err := spec.ImportSubnet(topology, *subnet.AvailabilityZone, *subnet.SubnetId, *subnet.CidrBlock); err != nil {
+			return err
+		}
 		spec.AppendAvailabilityZone(*subnet.AvailabilityZone)
 	}
 	return nil
 }
 
-// UseSubnets will update spec with subnets, it will call describeSubnets first,
+// UseSubnetsFromList will update spec with subnets, it will call describeSubnets first,
 // then pass resulting subnets to ImportSubnets
-func UseSubnets(provider api.ClusterProvider, spec *api.ClusterConfig, topology api.SubnetTopology, subnetIDs []string) error {
+func UseSubnetsFromList(provider api.ClusterProvider, spec *api.ClusterConfig, topology api.SubnetTopology, subnetIDs []string) error {
 	if len(subnetIDs) == 0 {
 		return nil
 	}
-
 	subnets, err := describeSubnets(provider, subnetIDs...)
 	if err != nil {
 		return err
 	}
-
 	return ImportSubnets(provider, spec, topology, subnets)
+}
+
+// UseSubnets will update spec with subnets, it will call describeSubnets first,
+// then pass resulting subnets to ImportSubnets
+func UseSubnets(provider api.ClusterProvider, spec *api.ClusterConfig) error {
+	if spec.VPC.ID != "" {
+		// ensure VPC gets imported and validated firt, if it's already set
+		if err := ImportVPC(provider, spec, spec.VPC.ID); err != nil {
+			return err
+		}
+	}
+	for topology := range spec.VPC.Subnets {
+		subnetIDs := []string{}
+		for _, subnet := range spec.VPC.Subnets[topology] {
+			subnetIDs = append(subnetIDs, subnet.ID)
+		}
+		if err := UseSubnetsFromList(provider, spec, topology, subnetIDs); err != nil {
+			return err
+		}
+	}
+	return nil
 }

